pkg/detectors/abuseipdb/abuseipdb.go

```diff
@@ -49,41 +59,51 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		}
 
 		if verify {
-			req, err := http.NewRequestWithContext(ctx, "GET", "https://api.abuseipdb.com/api/v2/check?ipAddress=118.25.6.39", nil)
-			if err != nil {
-				continue
-			}
-			req.Header.Add("Key", resMatch)
-			res, err := client.Do(req)
-			if err == nil {
-				bodyBytes, err := io.ReadAll(res.Body)
-				if err == nil {
-					bodyString := string(bodyBytes)
-					validResponse := strings.Contains(bodyString, `ipAddress`)
-					// errCode := strings.Contains(bodyString, `AbuseIPDB APIv2 Server.`)
-
-					defer res.Body.Close()
-					if res.StatusCode >= 200 && res.StatusCode < 300 {
-						if validResponse {
-							s1.Verified = true
-						} else {
-							s1.Verified = false
-						}
-					} else {
-						// This function will check false positives for common test words, but also it will make sure the key appears 'random' enough to be a real key.
-						if detectors.IsKnownFalsePositive(resMatch, detectors.DefaultFalsePositives, true) {
-							continue
-						}
-					}
-				}
-			}
+			client := s.getClient()
+			isVerified, verificationErr := verifyAbuseIPDB(ctx, client, resMatch)
+			s1.Verified = isVerified
+			s1.SetVerificationError(verificationErr, resMatch)
 		}
+
+		// This function will check false positives for common test words, but also it will make sure the key appears 'random' enough to be a real key.
+		if !s1.Verified && detectors.IsKnownFalsePositive(resMatch, detectors.DefaultFalsePositives, true) {
+			continue
+		}
+
 		results = append(results, s1)
 	}
 
 	return results, nil
 }
 
+func verifyAbuseIPDB(ctx context.Context, client *http.Client, resMatch string) (bool, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", "https://api.abuseipdb.com/api/v2/check?ipAddress=118.25.6.39", nil)
+	if err != nil {
+		return false, err
+	}
+	req.Header.Add("Key", resMatch)
+	res, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer res.Body.Close()
+	if res.StatusCode == http.StatusOK {
+		bodyBytes, err := io.ReadAll(res.Body)
+		if err != nil {
+			return false, err
+		}
+		validResponse := bytes.Contains(bodyBytes, []byte("ipAddress"))
+		if validResponse {
+			return true, nil
+		} else {
+			return false, nil
+		}
+	} else if res.StatusCode != http.StatusUnauthorized {
+		return false, fmt.Errorf("unexpected HTTP response status %d", res.StatusCode)
+	}
```

Comment:
Was this based on any documentation you read, or just reading the original implementation? (Or testing?)

Reply:
this was original implementation. i only tested to ensure it still worked, but didn't manually hit the endpoints myself.

Reply:
added docs now